code/modules/VR/vr_sleeper.dm

```diff
@@ -176,19 +171,61 @@
 
 /obj/machinery/vr_sleeper/proc/cleanup_vr_human()
 	if(vr_human)
-		vr_human.death(FALSE)
+		vr_human.vr_sleeper = null // Prevents race condition where a new human could get created out of order and set to null.
+		QDEL_NULL(vr_human)
+
+/obj/machinery/vr_sleeper/proc/emagNotify() 
+	if(vr_human)
+		vr_human.Dizzy(10)
 
 /obj/effect/landmark/vr_spawn //places you can spawn in VR, auto selected by the vr_sleeper during get_vr_spawnpoint()
 	var/vr_category = "default" //So we can have specific sleepers, eg: "Basketball VR Sleeper", etc.
-	var/vr_outfit = /datum/outfit/vr_basic
+	var/vr_outfit = /datum/outfit/vr
+
+/obj/effect/landmark/vr_spawn/Initialize()
+	. = ..()
+	build_spawnpoints()
```

Comment:
Rather than rebuilding the whole list repeatedly, just add/remove a given spawnpoint when it's created/destroyed.
```dm
LAZYADD(GLOB.vr_spawnpoints[vr_category], src)
LAZYREMOVE(GLOB.vr_spawnpoints[vr_category], src)
```

Reply:
I guess that works, I'll move the rebuild feature back in the sleepers though, it's handy when you var edit categories

Reply:
Some cases you want to trigger a rebuild

Reply:
Keeping the *ability* to rebuild is fine. One thing you could look at is conveyor belt code if you want to handle VVs automatically.

Reply:
Yeah I'll probably make more changes in the future, a lot of things planned that are oos for this pr

Reply:
Aight everything should be done